creator-node/src/routes/files.js

```diff
@@ -152,62 +151,47 @@ module.exports = function (app) {
 
     redisClient.incr('ipfsStandaloneReqs')
     const totalStandaloneIpfsReqs = parseInt(await redisClient.get('ipfsStandaloneReqs'))
-    logger.info(`IPFS Standalone Request - ${CID}`)
-    logger.info(`IPFS Stats - Standalone Requests: ${totalStandaloneIpfsReqs}`)
+    req.logger.info(`IPFS Standalone Request - ${CID}`)
+    req.logger.info(`IPFS Stats - Standalone Requests: ${totalStandaloneIpfsReqs}`)
 
     // If client has provided filename, set filename in header to be auto-populated in download prompt.
     if (req.query.filename) {
       res.setHeader('Content-Disposition', contentDisposition(req.query.filename))
     }
 
-    if (req.query.fromFS && req.query.fromFS === 'true') {
-      logger.info(`Retrieving directly from filesystem`)
-      // Retrieves the file directly from the filesystem rather than checking IPFS first
-
-      // Fire-and-forget rehydration
-      rehydrateIpfsFromFsIfNecessary(
-        req,
-        CID,
-        queryResults.storagePath
-      )
+    // Fire-and-forget rehydration
+    rehydrateIpfsFromFsIfNecessary(
+      req,
+      CID,
+      queryResults.storagePath
+    )
 
+    try {
```

Comment:
You should actually await the resp of streamFromFileSystem in this since returning it directly won't actually execute and errors won't be caught in the try

try {
  return await streamFromFileSystem
}

Reply:
https://repl.it/repls/DimgreyMajorNanocad#index.js